### PR TITLE
pdfstudio: remove libgccjit dependency

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

I don't think pdfstudio depends on libgccjit.  If it did, that would be a license violation.  Unlike the other libraries in gcc, libgccjit is licensed GPLv3 with no exceptions, since libgccjit contains the entire compiler.  PDF Studio is closed source.

I wasn't able to build pdfstudio2022 prior to this PR due to openjdk breakage (Edit: appears to be [this](https://github.com/NixOS/nixpkgs/pull/212363)).  However I did download and extract the `.deb` used, and ran this command, which produced no outputs:

```
find . -name \*.so -exec ldd {} \; | grep libgccjit
```

Maybe somebody confused `libgccjit` with (may it rest in peace) `libgcj`?

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/pdfstudio/default.nix [hunk 1]
```diff
@@ -49,7 +50,7 @@ in
       sha256 = "sha256-wQgVWz2kS+XkrqvCAUishizfDrCwGyVDAAU4Yzj4uYU=";
     };
     extraBuildInputs = [
-      libgccjit #for libstdc++.so.6 and libgomp.so.1
+      (lib.getLib gcc)  # for libstdc++.so.6 and libgomp.so.1
```
Comment (wegank) [inline pkgs/applications/misc/pdfstudio/default.nix:53]: Does `stdenv.cc.cc.lib` work here?

Comment (amjoseph-nixpkgs) [reply pkgs/applications/misc/pdfstudio/default.nix:53]: Testing this now.

Had to fix [the openjdk issue](https://github.com/NixOS/nixpkgs/pull/212363) first.

Comment (amjoseph-nixpkgs) [reply pkgs/applications/misc/pdfstudio/default.nix:53]: Yep, it does.

File: pkgs/applications/misc/pdfstudio/default.nix [hunk 2]
```diff
@@ -8,10 +8,12 @@
...
 # - pname is either "pdfstudio${year}" or "pdfstudioviewer".
 
-{ program ? "pdfstudio"
+{ lib
+, stdenv
+, program ? "pdfstudio"
 , year ? "2022"
 , fetchurl
-, libgccjit
+, gcc
```
Comment (wegank) [inline pkgs/applications/misc/pdfstudio/default.nix:16]:
```suggestion
```

**Comment by wegank** [comment]
@ofborg build pdfstudio2021 pdfstudio2022

**Pull request merged**